### ISSUE TITLE
[bug] edit of just alt text not accepted [feditext]

### ISSUE THREAD

**Issue by mirabilos** [superseriousbusiness/gotosocial#3817]
### Describe the bug with a clear and concise description of what the bug is.

When you edit a post and just change the alt text, the edit is not accepted as the post body stays the same; to accept it, one has to change a char in the post body as well.

### What's your GoToSocial Version?

0.18.0-rc2+git-1ee56d6 🦥 

### GoToSocial Arch

amd64 binary

### What you expected to happen?

While I’m sure that “not accepting an edit that doesn’t change anything” is a feature of sorts, it could take media changes (amount of media attached, which ones, changes to their alt text etc.) into account as well.

### Anything else we need to know?

This makes me think.

- Do changes of only CWs work? (It should, same as with media.)
- Do changes of posts that contain polls work? (It probably *should __not__* because doing so loses all votes.)

**Comment by tsmethurst** [comment]
I've just tested this myself to double check, and editing just alt text does work on the client I'm using at least (https://masto-fe.superseriousbusiness.org). I uploaded an image in a dm to myself with the alt text "whatever" then changed it to "whatever2". It also worked if I uploaded it first with no alt text, and then changed it to say "whatever". So I think this may be a client thing!

**Comment by mirabilos** [comment]
Hmm, according to this↓ FediText thinks it’s a 422 from GtS:

```
{
  "alertLocation" : {
    "file" : "ViewModels/ComposeStatusViewModel.swift",
    "function" : "post(viewModel:inReplyToId:)",
    "line" : 404
  },
  "app" : {
    "build" : "130",
    "bundle" : "com.gotgoat.metatext",
    "version" : "1.7.2"
  },
  "error" : {
    "apiCapabilities" : {
      "features" : [

      ],
      "flavor" : "gotosocial",
      "version" : "0.18.0-rc3+git-8829ee1"
    },
    "apiError" : {
      "error" : "Unprocessable Entity: status was not changed"
    },
    "method" : "PUT",
    "requestLocation" : {
      "file" : "ServiceLayer/IdentityService.swift",
      "function" : "put(id:statusComponents:)",
      "line" : 385
    },
    "statusCode" : 422,
    "url" : "https://toot.XXX.org/api/v1/statuses/01JMJS1R3R5TCA482H7CVGG3NE"
  },
  "message" : "PUT https://toot.XXX.org/api/v1/statuses/01JMJS1R3R5TCA482H7CVGG3NE\n\nUnprocessable Entity: status was not changed",
  "system" : {
    "idiom" : "phone",
    "model" : "iPhone",
    "name" : "iOS",
    "version" : "18.3.1"
  },
  "title" : "AnnotatedAPIError",
  "type" : "AnnotatedAPIError"
}
```

**Comment by tsmethurst** [comment]
Maybe Feditext isn't submitting the alt text change 🤔 I'm not sure. Maybe worth raising an issue on their repo and linking back to this one.

**Comment by tsmethurst** [comment]
I just tested on Tusky as well and it works there too (just adding or changing alt text, that is).

**Comment by mirabilos** [comment]
Submitted as https://github.com/feditext/feditext/issues/480 though if I submit the change of both the body and alt text, the alt text is indeed changed. But perhaps FediText doesn’t submit the alt text change if the body text isn’t changed, or something.

Thanks for taking the effort to check on other clients!

**Comment by tsmethurst** [comment]
No worries!

**Comment by NyaaaWhatsUpDoc** [comment]
given (so far) this only seems repeatable in feditext but not in Tusky, i'm going to leave investigation first up to feditext, and depending on if they find anything i'll take a look in our edit processing code :). just a ping to @VyrCossont so you're aware of this, since you're the feditext gotosocial crossover dev :p

**Comment by daenney** [comment]
We've migrated to Codeberg. Your issue has been imported as https://codeberg.org/superseriousbusiness/gotosocial/issues/3817.
